### PR TITLE
using payload and err variables declared in function signature

### LINKED ISSUE #140: jws.VerifyWithJKU Passes on invalid signature
I was using v0.9.0,  `jws.VerifyWithJKU()` validation worked fine. But once I updated to the newer version v0.9.1, it no longer validates the signature, returns no error. This is a serious issue. Could I get any help regarding this? 

Thank you.

### PR DESCRIPTION
**Pull request by leandrorondon**
fix #140

### CODE REVIEW

File: jws/jws.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func VerifyWithJWKSet(buf []byte, keyset *jwk.Set, keyaccept JWKAcceptFunc) (pay
 			continue
 		}
 
-		payload, err := VerifyWithJWK(buf, key)
+		payload, err = VerifyWithJWK(buf, key)
```
Comment (leighmcculloch) [inline jws/jws.go:366]: It's interesting that the tests pass with and without this change. Could you add a test as part of this fix?

**Comment by leandrorondon** [comment]
> It's interesting that the tests pass with and without this change. Could you add a test as part of this fix?

> This fix looks good to me. This also seems like a reasonably serious issue that the error is not being propagated. Is it possible to include this in a release?

@leighmcculloch Thanks for the feedback.

It's the first time I contribute to an open-source project :)
Can I make the requested changes in this same PR?

**Comment by leighmcculloch** [comment]
@leandrorondon I'm not a contributor to this project so I can't speak for @lestrrat. I would think adding some tests would be good.

**Comment by lestrrat** [comment]
Sorry for the delay here. Yeah, I think a test would be nice.

**Comment by lestrrat** [comment]
Okay, so I finally got around to looking at the problem (sorry for warnocking #140). I don't think this PR fixes anything, and #140 certainly doesn't have anything to do with recent changes to this particular piece of code.

As for this particular piece of code, I can certainly tidy it up and I will do so

However, I don't think this fixes #140.

**Comment by lestrrat** [comment]
oh wait, I'm sorry, I spoke too soon. Sorry @leandrorondon . Will update soon....

**Comment by lestrrat** [comment]
I think #145 fixes is more "properly"

**Comment by lestrrat** [comment]
closed by #145

**Pull request closed without merging**